packages/@aws-cdk/aws-sqs/lib/send-message-task.ts

```diff
@@ -0,0 +1,128 @@
+import iam = require('@aws-cdk/aws-iam');
+import stepfunctions = require('@aws-cdk/aws-stepfunctions');
+import cdk = require('@aws-cdk/cdk');
+import { IQueue } from './queue-ref';
+
+/**
+ * Properties for SendMessageTask
+ */
+export interface SendMessageTaskProps extends stepfunctions.BasicTaskProps {
+  /**
+   * The topic to send a message to to
+   */
+  queue: IQueue;
+
+  /**
+   * The message body to send to the queue.
+   *
+   * Exactly one of `messageBody` and `messageBodyPath` is required.
+   */
+  messageBody?: string;
+
+  /**
+   * JSONPath for the message body to send to the queue.
+   *
+   * Exactly one of `messageBody` and `messageBodyPath` is required.
+   */
+  messageBodyPath?: string;
+
+  /**
+   * The length of time, in seconds, for which to delay a specific message.
+   *
+   * Valid values are 0-900 seconds.
+   *
+   * @default Default value of the queue is used
+   */
+  delaySeconds?: number;
+
+  /**
+   * JSONPath expression for delaySeconds setting
+   *
+   * @default Default value of the queue is used
+   */
+  delaySecondsPath?: string;
+
+  /**
+   * The token used for deduplication of sent messages.
+   *
+   * @default Use content-based deduplication
+   */
+  messageDeduplicationId?: string;
+
+  /**
+   * JSONPath expression for deduplication ID
+   *
+   * @default Use content-based deduplication
+   */
+  messageDeduplicationIdPath?: string;
+
+  /**
+   * The tag that specifies that a message belongs to a specific message group.
+   *
+   * Required for FIFO queues. FIFO ordering applies to messages in the same message
+   * group.
+   *
+   * @default No group ID
+   */
+  messageGroupId?: string;
+
+  /**
+   * JSONPath expression for message group ID
+   *
+   * @default No group ID
+   */
+  messageGroupIdPath?: string;
+}
+
+/**
+ * A StepFunctions Task to send a message to an SQS Queue
+ */
+export class SendMessageTask extends stepfunctions.Task {
```

Comment:
Should we add `queue.toStepFunction()` as a convenience/discovery mechanism? 

@skinny85, this is similar to what we have for pipelines (`toCodePipeline()`).